dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -463,18 +463,27 @@ public WorkflowVersion getWorkflowVersionById(@Parameter(hidden = true, name = "
         @Parameter(name = "workflowId", description = "id of the workflow", required = true, in = ParameterIn.PATH) @PathParam("workflowId") Long workflowId,
         @Parameter(name = "workflowVersionId", description = "id of the workflow version", required = true, in = ParameterIn.PATH) @PathParam("workflowVersionId") Long workflowVersionId,
         @Parameter(name = "include", description = VERSION_INCLUDE_MESSAGE, in = ParameterIn.QUERY) @QueryParam("include") String include) {
-        Workflow workflow = workflowDAO.findById(workflowId);
-        checkNotNullEntry(workflow);
-        checkCanExamine(user, workflow);
-
-        WorkflowVersion workflowVersion = this.workflowVersionDAO.findById(workflowVersionId);
-        if (workflowVersion == null) {
-            throw new CustomWebApplicationException("Version " + workflowVersionId + " does not exist for this workflow", HttpStatus.SC_NOT_FOUND);
-        }
+        WorkflowVersion workflowVersion =
+            getWorkflowVersion(user, workflowId, workflowVersionId);
         initializeAdditionalFields(include, workflowVersion);
         return workflowVersion;
     }
 
+    @GET
```

Comment:
Not sure how I feel about this since we use the "include" pattern elsewhere (such as https://dockstore.org/api/static/swagger-ui/index.html#/workflows/getWorkflow ), we do use this pattern on Organizations though. 
 
This is also a special case for performance (i.e. we shouldn't get a large number of these endpoints to get individual fields of a workflow), so I'm on the fence.

Reply:
We have this pattern for workflow version as well, although admittedly they are for collections, not a single property:

* [Orcid Authors](https://dockstore.org/api/static/swagger-ui/index.html#/workflows/getWorkflowVersionOrcidAuthors)
* [Source files](https://dockstore.org/api/static/swagger-ui/index.html#/workflows/getWorkflowVersionsSourcefiles)

It seemed less disruptive to do it this way. We have a whole slew of endpoints that return a workflow or workflows, over a dozen, most of which don't yet have the `include` query parameter. And then there are also a few WorkflowVersion endpoints, the container, and the tag endpoints. And this property is a real outlier, size-wise. Finally, I like it because the type indicates that the property isn't present.

I could add description to the include of [workflowVersion endpoint](https://dockstore.org/api/static/swagger-ui/index.html#/workflows/getWorkflowVersionById) and any workflow endpoints that have an include; everywhere else we null out the description.

I'm on the fence too. :) Any strong opinions by anybody?

Reply:
> I could add description to the include of [workflowVersion endpoint](https://dockstore.org/api/static/swagger-ui/index.html#/workflows/getWorkflowVersionById) and any workflow endpoints that have an include; everywhere else we null out the description.

I didn't even notice that the workflow version one also had an include option! How do we rationalise which fields to put in `include` and which ones to make an extra endpoint? (I realize that we haven't been terribly consistent, but I'm trying to think of a rule of thumb)

Reply:
> I didn't even notice that the workflow version one also had an include option! How do we rationalise which fields to put in `include` and which ones to make an extra endpoint? (I realize that we haven't been terribly consistent, but I'm trying to think of a rule of thumb)

I've been poking around and struggling to find a good answer for this. 

* Some of the [Google API](https://developers.google.com/books/docs/v1/reference/volumes/get) uses a `projection` query parameter, analogous to our `include` query param.
* I had a hard time finding an example of this, but I did find a singular "sub-path" in the GitHub API [here](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-contextual-information-for-a-user), so there is precedent for a non-collection sub-resource; I was worried that "extra endpoints" were only for collections.

I like the way I did this for implementation reasons -- otherwise I'll have to go through a couple of dozen endpoints and null out the description field. Or have the `getWorkflowVersion` endpoint return a different class, e.g., `WorkflowVersion2`, which includes the description. Or be inconsistent and null out the `description` field on the expensive calls only.

But implementation shouldn't be the overriding factor... I think I lean towards this approach without completely being able to justify it. :(

Reply:
#5595 for followup